File: led.zig
const std = @import("std");
const wpi = @cImport({
    @cInclude("wiringPi.h");
});

const LEDPIN = @as(c_int, 0);
const DELAY = @as(c_uint, 500);

pub fn main() !void {
    try wpi.wiringPiSetup();
    // if (wpi.wiringPiSetup() == -1) {
    //     std.debug.print("Setup wiringPi failed !", .{});
    // }
    try wpi.pinMode(LEDPIN, wpi.OUTPUT);
    while (true) {
        try wpi.digitalWrite(LEDPIN, wpi.LOW);
        std.debug.print("...LED on\n", .{});
        try wpi.delay(DELAY);

        try wpi.digitalWrite(LEDPIN, wpi.HIGH);
        std.debug.print("LED off...\n", .{});
        try wpi.delay(DELAY);
    }
}

File: led.c
#include <stdio.h>
#include <wiringPi.h> 

#define LedPin 0

int main() {
   if (wiringPiSetup() == -1) {
      printf("setup wiringPi failed !");
      return 1;
   }
   pinMode(LedPin, OUTPUT);
   while(1) {
      digitalWrite(LedPin, LOW);
      printf("...LED on\n");
      delay(500);
      digitalWrite(LedPin, HIGH);
      printf("Led off...\n");
      delay(500);
   }
   return 0;
}
